toolkits/google/arcade_google/tools/sheets.py

```diff
@@ -0,0 +1,144 @@
+    title: Annotated[str, "The title of the new spreadsheet"] = "Untitled spreadsheet",
+    data: Annotated[
+        Optional[str],
+        "The data to write to the spreadsheet. A JSON string "
+        "(property names enclosed in double quotes) representing a dictionary that "
+        "maps row numbers to dictionaries that map column letters to cell values. "
+        "For example, data[23]['C'] would be the value of the cell in row 23, column C. "
+        "Type hint: dict[int, dict[str, Union[int, float, str, bool]]]",
+    ] = None,
+) -> Annotated[dict, "The created spreadsheet's id and title"]:
+    """Create a new spreadsheet with the provided title and data in its first sheet
+
+    Returns the newly created spreadsheet's id and title
+    """
+    service = build_sheets_service(context.get_auth_token_or_empty())
+
+    try:
+        sheet_data = SheetDataInput(data=data)  # type: ignore[arg-type]
+    except Exception as e:
+        msg = "Invalid JSON or unexpected data format for parameter `data`"
+        raise RetryableToolError(
+            message=msg,
+            additional_prompt_content=f"{msg}: {e}",
+            retry_after_ms=100,
+        )
+
+    spreadsheet = Spreadsheet(
+        properties=SpreadsheetProperties(title=title),
+        sheets=[create_sheet(sheet_data)],
+    )
+
+    body = spreadsheet.model_dump()
+
+    response = (
+        service.spreadsheets()
+        .create(body=body, fields="spreadsheetId,spreadsheetUrl,properties/title")
+        .execute()
+    )
+
+    return {
+        "title": response["properties"]["title"],
+        "spreadsheetId": response["spreadsheetId"],
+        "spreadsheetUrl": response["spreadsheetUrl"],
+    }
+
+
+@tool(
+    requires_auth=Google(
+        scopes=["https://www.googleapis.com/auth/drive.file"],
+    )
+)
+async def get_spreadsheet(
+    context: ToolContext,
+    spreadsheet_id: Annotated[str, "The id of the spreadsheet to get"],
+) -> Annotated[
+    dict,
+    "The spreadsheet properties and data for all sheets in the spreadsheet",
+]:
+    """
+    Get the user entered values and formatted values for all cells in all sheets in the spreadsheet
+    along with the spreadsheet's properties
+    """
+    service = build_sheets_service(context.get_auth_token_or_empty())
+    response = (
+        service.spreadsheets()
+        .get(
+            spreadsheetId=spreadsheet_id,
+            includeGridData=True,
+            fields="spreadsheetId,spreadsheetUrl,properties/title,sheets/properties,sheets/data/rowData/values/userEnteredValue,sheets/data/rowData/values/formattedValue,sheets/data/rowData/values/effectiveValue",
+        )
+        .execute()
+    )
+    return parse_get_spreadsheet_response(response)
+
+
+@tool(
+    requires_auth=Google(
+        scopes=["https://www.googleapis.com/auth/drive.file"],
+    )
+)
+def write_to_cell(
+    context: ToolContext,
+    spreadsheet_id: Annotated[str, "The id of the spreadsheet to write to"],
```

Comment:
In the future we can also use `spreadsheet_name` if there is a Google Drive tool added for getting the name of spreadsheet by ID